projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLDataFetchers.java

```diff
@@ -227,6 +236,49 @@ private List<V2DataJob> populateDeployments(
     return allDataJob;
   }
 
+  private Map<String, JobDeploymentStatus> readJobDeploymentsFromK8s() {
+    return deploymentService.readDeployments().stream()
+        .collect(Collectors.toMap(JobDeploymentStatus::getDataJobName, cronJob -> cronJob));
+  }
+
+  private Map<String, JobDeploymentStatus> readJobDeploymentsFromDb() {
+    var deployments =
+        StreamSupport.stream(actualJobDeploymentRepository.findAll().spliterator(), false)
+            .collect(Collectors.toMap(ActualDataJobDeployment::getDataJobName, cronjob -> cronjob));
+
+    return deployments.entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey, entry -> convertToJobDeploymentStatus(entry.getValue())));
+  }
+
+  private JobDeploymentStatus convertToJobDeploymentStatus(
```

Comment:
The following methods should be part of the DeploymentModelConverter

Reply:
Moved, and removed one of the methods, as there was an already existing method doing the same checks. Thanks for the review

Reply:
Also, I see that in the Deployment Controller, https://github.com/vmware/versatile-data-kit/blob/main/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/DataJobsDeploymentController.java#L134, when we read from the DB, we return `DataJobDeploymentStatus`, while in the GraphQL logic, we return `JobDeploymentStatus` everywhere. Any idea why this is the case?

Reply:
No idea :)

Reply:
Cool...